src/main/scala/scalismo/mesh/MeshOperations.scala

```diff
@@ -99,12 +99,28 @@ class TriangleMesh3DOperations(private val mesh: TriangleMesh3D) {
    *
    */
   def toBinaryImage: ScalarImage[_3D] = {
+
+    val meshOps = mesh.operations
+
     def inside(pt: Point[_3D]): Short = {
-      val closestMeshPt = mesh.pointSet.findClosestPoint(pt)
-      val dotprod = mesh.vertexNormals(closestMeshPt.id) dot (closestMeshPt.point - pt)
+      val (closestPoint, normalAtClosestPoint) = meshOps.closestPointOnSurface(pt) match {
+        case ClosestPointInTriangle(closestPoint, dist, triangleId, bcc) => {
+          (closestPoint, mesh.vertexNormals.onSurface(triangleId, bcc))
+        }
+        case ClosestPointOnLine(closestPoint, _, (id1, id2), bc) => {
+          val normalPt1 = mesh.vertexNormals(id1)
+          val normalPt2 = mesh.vertexNormals(id2)
+          val averagedNormal = (normalPt1 * bc) + (normalPt2 * (1.0 - bc))
+          (closestPoint, averagedNormal / averagedNormal.norm)
+        }
+        case _ => {
```

Comment:
@Andreas-Forster As this is only a stylistic issue, which will anyway be addressed when we fix issue #201, I will merge the PR.